non-prod/us-east-1/qa/mysql/terragrunt.hcl

```diff
@@ -1,30 +1,34 @@
-locals {
-  # Automatically load environment-level variables
-  environment_vars = read_terragrunt_config(find_in_parent_folders("env.hcl"))
+# ---------------------------------------------------------------------------------------------------------------------
+# TERRAGRUNT CONFIGURATION
+# This is the configuration for Terragrunt, a thin wrapper for Terraform that helps keep your code DRY and
+# maintainable: https://github.com/gruntwork-io/terragrunt
+# ---------------------------------------------------------------------------------------------------------------------
 
-  # Extract out common variables for reuse
-  env = local.environment_vars.locals.environment
-}
-
-# Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
-# working directory, into a temporary folder, and execute your Terraform commands in that folder.
+# We override the terraform block source attribute here just for the QA environment to show how you would deploy a
+# different version of the module in a specific environment.
 terraform {
-  source = "git::git@github.com:gruntwork-io/terragrunt-infrastructure-modules-example.git//mysql?ref=v0.4.0"
+  source = "${include.envcommon.locals.base_source_url}?ref=v0.4.0"
```

Comment:
NIT: if you have the `terraform { source = "..." }` block in each child `terragrunt.hcl` anyway, is there a point in putting it into the `envcommon` files?

Reply:
I only have this in the `qa` config. Note that it is missing from [prod](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/blob/yori-dry-tg/prod/us-east-1/prod/mysql/terragrunt.hcl) and [stage](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/blob/yori-dry-tg/non-prod/us-east-1/stage/mysql/terragrunt.hcl).